market/mysterium/response.go

```diff
@@ -24,22 +24,20 @@ import (
 	"net/http"
 )
 
-type errorResponse struct {
-	Error string `json:"error"`
-}
-
 // ParseResponseError checks the respose for correctness and return error if it is invalid
-func ParseResponseError(response *http.Response) error {
+func ParseResponseError(response *http.Response) (err error) {
 	if response.StatusCode < 200 || response.StatusCode >= 300 {
-		var error errorResponse
+		var errorMsg []byte
 
-		err := ParseResponseJSON(response, &error)
-		if err != nil {
-			return err
+		if response.Body != nil {
+			errorMsg, err = ioutil.ReadAll(response.Body)
+			if err != nil {
+				return err
```

Comment:
I would suggest wrapping this error too into our "server response invalid: ....". Because in case of failed body read you will get for example "EOF" only with no context where it failed.
A common message "server respone invalid: ...." should be better for both cases